src/components/FlagsAutocomplete/FlagsAutocomplete.tsx

```diff
@@ -0,0 +1,227 @@
+  const {
+    anchorEl,
+    onSelectCountry,
+    excludedCountries,
+    onlyCountries,
+    langOfCountryName,
+    continents,
+    preferredCountries,
+    className,
+    flagSize,
+    onClose
+  } = props
+
+  // Idem for the translations
+  const displayNames = React.useMemo(() => {
+    return getDisplayNames(langOfCountryName)
+  }, [langOfCountryName])
+
+  const ISO_CODES_SORTED = sortAlphabeticallyCountryCodes(
+    ISO_CODES,
+    displayNames
+  )
+
+  const countriesFiltered = filterCountries(ISO_CODES_SORTED, {
+    onlyCountries,
+    excludedCountries,
+    continents,
+    preferredCountries
+  })
+
+  const countriesFilteredOptions: MuiTelAutocompleteOption[] =
+    countriesFiltered.map((countryCode) => {
+      return {
+        countryCode,
+        callingCode: COUNTRIES[countryCode]?.[0] as string,
+        displayName: displayNames.of(countryCode) ?? countryCode
+      }
+    })
+
+  const filterOptions = (
+    options: MuiTelAutocompleteOption[],
+    { inputValue }: FilterOptionsState<MuiTelAutocompleteOption>
+  ) => {
+    if (inputValue === '') {
+      return options
+    }
+
+    return matchSorter(options, inputValue, {
+      keys: ['callingCode', 'countryCode', 'displayName']
+    })
+  }
+
+  const getOptionLabel = (option: string | MuiTelAutocompleteOption) => {
+    if (typeof option === 'string') {
+      return ''
+    }
+
+    return option.countryCode
+  }
+
+  return (
+    <Styled.FlagsAutocompletePopper
+      anchorEl={anchorEl}
+      className={`MuiTelInput-FlagsAutocomplete-Popover ${className || ''}`}
+      id="muitelinput-flagsautocomplete"
+      open={Boolean(anchorEl)}
+      placement="bottom-start"
+    >
+      <ClickAwayListener onClickAway={onClose}>
+        <Autocomplete
+          autoHighlight
+          filterOptions={filterOptions}
+          freeSolo
+          getOptionLabel={getOptionLabel}
+          onChange={(event, newValue, reason) => {
+            if (
+              event.type === 'keydown' &&
+              (event as React.KeyboardEvent).key === 'Backspace' &&
+              reason === 'removeOption'
+            ) {
+              return
+            }
+
+            if (newValue !== null) {
+              if (typeof newValue !== 'string') {
+                onSelectCountry(newValue.countryCode)
+              }
+            }
+          }}
+          openOnFocus
+          onClose={onClose}
+          options={countriesFilteredOptions}
+          PopperComponent={PopperComponent}
+          renderInput={(params) => {
+            return (
+              <Styled.Input
+                // eslint-disable-next-line jsx-a11y/no-autofocus
+                autoFocus
+                className="MuiTelInput-FlagsAutocomplete-Input"
+                inputProps={{
+                  ...params.inputProps,
+                  'data-testid': 'flagsautocomplete-input'
+                }}
+                placeholder="Search for a country"
```

Comment:
So not I18n possible ?

Reply:
Would you rather that we remove the placeholder for now? I think a follow-up PR to add support for i18n would make sense.

Reply:
got it